cli/azd/cmd/deploy.go

```diff
@@ -93,6 +94,7 @@ type deployAction struct {
 
 func newDeployAction(
 	flags *deployFlags,
+	azdCtx *azdcontext.AzdContext,
```

Comment:
Since the action will now be instantiated at the correct time we can now inject the `azdcontext.AzdContxt` like we wanted to do originally.

Reply:
I think this is a good win since we really should know about the `azd` project layout when actions run. The composite action construction was throwing us off, but now this can be restored.